agent/agent_endpoint.go

```diff
@@ -1254,14 +1227,14 @@ func (s *HTTPHandlers) AgentDeregisterService(resp http.ResponseWriter, req *htt
 
 	sid.Normalize()
 
-	if !s.validateRequestPartition(resp, &sid.EnterpriseMeta) {
-		return nil, nil
-	}
-
 	if err := s.agent.vetServiceUpdateWithAuthorizer(authz, sid); err != nil {
 		return nil, err
 	}
 
+	if !s.validateRequestPartition(resp, &sid.EnterpriseMeta) {
+		return nil, nil
+	}
+
```

Comment:
Is there a reason this has been moved?

Partitions are an enterprise feature and we intentionally check for this before ACLs

Reply:
I honestly don’t remember moving that or why I might have done that. I’ll fix that as soon as I can.